src/admin/resolvers/mutations/ScheduledItem.ts

```diff
@@ -117,3 +122,61 @@ export async function createScheduledItem(
     throw new Error(error);
   }
 }
+
+export async function rescheduleScheduledItem(
+  parent,
+  { data },
+  context: IContext
+): Promise<ScheduledItem> {
+  // Need to fetch the item first to check access privileges.
+  // Note that we do not worry here about an extra hit to the DB
+  // as load on this service will be low.
+  const item = await context.db.scheduledItem.findUnique({
+    where: { externalId: data.externalId },
+  });
+
+  if (!item) {
+    throw new NotFoundError(
+      `Item with ID of '${data.externalId}' could not be found.`
+    );
+  }
+
+  // Check if the user can execute this mutation.
+  if (!context.authenticatedUser.canWriteToSurface(item.scheduledSurfaceGuid)) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
+
+  try {
+    const rescheduledItem = await dbRescheduleScheduledItem(
+      context.db,
+      data,
+      context.authenticatedUser.username
+    );
+
+    context.emitScheduledCorpusItemEvent(
+      ScheduledCorpusItemEventType.RESCHEDULE,
+      rescheduledItem
+    );
+
+    return rescheduledItem;
+  } catch (error) {
+    // If it's the duplicate scheduling constraint, catch the error
+    // and send a user-friendly one to the client instead.
+    if (
+      error instanceof PrismaClientKnownRequestError &&
+      error.code === 'P2002'
+    ) {
+      throw new UserInputError(
+        `This story is already scheduled to appear on ${data.scheduledDate.toLocaleString(
+          'en-US',
```

Comment:
Wondering if the `en-US` date format would be confusing for the non North American curators?

Reply:
tests are in there! github just collapses the file because it's a big one. 😄 

as for the date format - maybe. (for better or for worse) our entire interface is US-centric. maybe down the road we can figure out how to display the date based on the current user's locale, but today is not that day.